packages/jest-environment/src/index.ts

```diff
@@ -22,15 +22,23 @@ export type EnvironmentContext = {
 // TODO: type this better: https://nodejs.org/api/modules.html#modules_the_module_wrapper
 type ModuleWrapper = (...args: Array<unknown>) => unknown;
 
+export function hasModuleWrapper(
+  obj: unknown,
+): obj is {[ScriptTransformer.EVAL_RESULT_VARIABLE]: ModuleWrapper} {
+  return (
+    typeof obj === 'object' &&
+    !!obj &&
+    obj.hasOwnProperty(ScriptTransformer.EVAL_RESULT_VARIABLE)
+  );
+}
+
 export declare class JestEnvironment {
   constructor(config: Config.ProjectConfig);
   constructor(config: Config.ProjectConfig, context: EnvironmentContext);
   global: Global.Global;
   fakeTimers: FakeTimers<unknown> | null;
   moduleMocker: ModuleMocker | null;
-  runScript(
```

Comment:
in case the comment directly on the commit was lost: I'd keep this as it was, and move the check into `jest-environment-jsdom` (or just do `as any` there). It's an error from the environments if they don't return `{[ScriptTransformer.EVAL_RESULT_VARIABLE]: ModuleWrapper} | null`, and it would be great if the typings told developers that instead of failing at runtime